packages/flutter_tools/lib/src/localizations/gen_l10n.dart

```diff
@@ -1189,10 +1189,15 @@ class LocalizationsGenerator {
     );
 
     final List<LocaleInfo> allLocales = _allBundles.locales.toList()..sort();
-    final String fileName = outputFileName.split('.')[0];
+    final int extensionIndex = outputFileName.indexOf('.');
+    // TBD error checking and reporting:
```

Comment:
Please include a check (or checks) and a `throwToolExit` that explains the problem.

Reply:
I made it throw L10nException as documented in LocalizationsGenerator:

https://github.com/flutter/flutter/blob/f4f23ecb599b6ac28a5affd8fededf3ab2aaaf77/packages/flutter_tools/lib/src/localizations/gen_l10n.dart#L655-L656